src/components/organism/SwapTokens/index.tsx

```diff
@@ -104,87 +339,226 @@ const SwapTokens = () => {
+
+      assetTokensInPoolTokenPairsArray.push(TokenSelection.NativeToken);
+
+      const assetTokensNotInPoolTokenPairsArray = assetTokens.filter((item: any) =>
+        assetTokensInPoolTokenPairsArray.includes(item.assetTokenMetadata.symbol)
+      );
+
+      return assetTokensNotInPoolTokenPairsArray;
+    }
+  };
+
+  const handleSwap = async () => {
+    if (api) {
+      const tokenA = formatInputTokenValue(tokenAValueForSwap.tokenValue, selectedTokens.tokenA.decimals);
+      const tokenB = formatInputTokenValue(tokenBValueForSwap.tokenValue, selectedTokens.tokenB.decimals);
+      if (selectedTokens.tokenA.tokenSymbol === TokenSelection.NativeToken) {
+        if (selectedTokens.tokenB.tokenId) {
+          if (inputEdited.inputType === InputEditedType.exactIn) {
+            await swapNativeForAssetExactIn(
+              api,
+              selectedTokens.tokenB.tokenId,
+              selectedAccount,
+              tokenA,
+              tokenB,
+              false,
+              dispatch
+            );
+          } else if (inputEdited.inputType === InputEditedType.exactOut) {
+            if (selectedTokens.tokenB.tokenId) {
+              await swapNativeForAssetExactOut(
+                api,
+                selectedTokens.tokenB.tokenId,
+                selectedAccount,
+                tokenA,
+                tokenB,
+                false,
+                dispatch
+              );
+            }
+          }
+        }
+      } else if (selectedTokens.tokenB.tokenSymbol === TokenSelection.NativeToken) {
+        if (selectedTokens.tokenA.tokenId) {
+          if (inputEdited.inputType === InputEditedType.exactIn) {
+            await swapNativeForAssetExactIn(
+              api,
+              selectedTokens.tokenA.tokenId,
+              selectedAccount,
+              tokenB,
+              tokenA,
+              true,
+              dispatch
+            );
+          } else if (inputEdited.inputType === InputEditedType.exactOut) {
+            await swapNativeForAssetExactOut(
+              api,
+              selectedTokens.tokenA.tokenId,
+              selectedAccount,
+              tokenB,
+              tokenA,
+              true,
+              dispatch
+            );
+          }
+        }
+      } else if (
+        selectedTokens.tokenA.tokenSymbol !== TokenSelection.NativeToken &&
+        selectedTokens.tokenB.tokenSymbol !== TokenSelection.NativeToken
+      ) {
+        if (selectedTokens.tokenA.tokenId && selectedTokens.tokenB.tokenId) {
+          if (inputEdited.inputType === InputEditedType.exactIn) {
+            await swapAssetForAssetExactIn(
+              api,
+              selectedTokens.tokenA.tokenId,
+              selectedTokens.tokenB.tokenId,
+              selectedAccount,
+              tokenA,
+              tokenB,
+              dispatch
+            );
+          } else if (inputEdited.inputType === InputEditedType.exactOut) {
```

Comment:
You can put child condition (if condition below) in same line as else if, or another reason exists?